internal/allocator/allocator_test.go

```diff
@@ -520,6 +531,29 @@ func TestAssignment(t *testing.T) {
 			ips:     []string{"1.2.6.10"},
 			wantErr: true,
 		},
+		{
+			desc:   "attempt to assign ip from pool pinned with namespace and service label",
```

Comment:
can we add two tests that should fail because the labels are compatible but the ns is not and viceversa?

Reply:
done